fluxsauce/site_audit
# Added database fragmentation check

Issue #51 
To test,

``` bash
 ❯ drush @d8 ad --detail                                [13:57:58]
Database: 83%
  Total size: Determine the size of the database.
    Total size: 7.41MB
  Collations: Check to see if there are any tables that aren't using UTF-8.
    Every table is using UTF-8.
  Storage Engines: Check to see if there are any tables that aren't using InnoDB.
    Every table is using InnoDB.
  Tables with at least 1000 rows: Return list of all tables with at least 1000 rows in the database.
    No tables with less than 1000 rows.
  Database Fragmentation: Reports all the tables with fragmentation greater than 5%
    Table Name: Fragmentation Percentage
    ---------------------
    cache_discovery: 400
      Run "OPTIMIZE TABLE" on the fragmented tables. Refer to https://dev.mysql.com/doc/refman/5.5/en/optimize-table.html for more details.
```

``` bash
 ❯ mysql -u root -p                                     [13:58:05]
Enter password: 
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 128
Server version: 10.0.20-MariaDB-log MariaDB Server

Copyright (c) 2000, 2015, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]> OPTIMIZE TABLE drupal8.cache_discovery;
+-------------------------+----------+----------+-------------------------------------------------------------------+
| Table                   | Op       | Msg_type | Msg_text                                                          |
+-------------------------+----------+----------+-------------------------------------------------------------------+
| drupal8.cache_discovery | optimize | note     | Table does not support optimize, doing recreate + analyze instead |
| drupal8.cache_discovery | optimize | status   | OK                                                                |
+-------------------------+----------+----------+-------------------------------------------------------------------+
2 rows in set (2.63 sec)

MariaDB [(none)]> exit
Bye
```

``` bash
 ❯ drush @d8 ad --detail                                [13:58:29]
Database: 100%
  No action required.                                   [success]
  Total size: Determine the size of the database.
    Total size: 7.55MB
  Collations: Check to see if there are any tables that aren't using UTF-8.
    Every table is using UTF-8.
  Storage Engines: Check to see if there are any tables that aren't using InnoDB.
    Every table is using InnoDB.
  Tables with at least 1000 rows: Return list of all tables with at least 1000 rows in the database.
    No tables with less than 1000 rows.
  Database Fragmentation: Reports all the tables with fragmentation greater than 5%
    None of the tables has fragmentation of more than 5%
```
